Source/JavaScriptCore/assembler/LinkBuffer.h

```diff
@@ -39,10 +39,13 @@
 #include "MacroAssemblerCodeRef.h"
 #include <wtf/DataLog.h>
 #include <wtf/FastMalloc.h>
+#include <wtf/NeverDestroyed.h>
 #include <wtf/Noncopyable.h>
 
 namespace JSC {
 
+DECLARE_ALLOCATOR_WITH_HEAP_IDENTIFIER(BranchCompactionLinkBuffer);
```

Comment:
This should stay in the .cpp since the class is wholly contained there.

Reply:
Get this error if I keep I keep it in the cpp file.

```
JavaScriptCore GenerateTAPI JavaScriptCore.tbd
warning: violations found for arm64e
error: no declaration found for exported symbol 'JSC::BranchCompactionLinkBufferMalloc::debugHeap()' in dynamic library
WebGPU CompileC Adapter.o
WebGPU CompileC CommandEncoder.o
```

Reply:
If it stays in the class, it needs to have visibility hidden, because otherwise you're exporting the debugHeap() symbol without putting a declaration in headers for a client to import.

Reply:
Yep so you should pass `WTF_PRIVATE` as the second parameter to the DECLARE macro.

Reply:
jinx

Reply:
Changed to `DECLARE_ALLOCATOR_WITH_HEAP_IDENTIFIER_AND_EXPORT(BranchCompactionLinkBuffer, WTF_INTERNAL);
` and it builds successfully now.